web/src/main/java/org/springframework/security/web/savedrequest/SavedCookie.java

```diff
@@ -31,22 +31,28 @@ public class SavedCookie implements Serializable {
 
 	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
 
-	private final java.lang.String name;
+	private final String name;
 
-	private final java.lang.String value;
+	private final String value;
 
-	private final java.lang.String comment;
+	private final String comment;
 
-	private final java.lang.String domain;
+	private final String domain;
 
 	private final int maxAge;
 
-	private final java.lang.String path;
+	private final String path;
 
 	private final boolean secure;
 
 	private final int version;
 
+	/**
+	 * @deprecated use
+	 * {@link SavedCookie#SavedCookie(String, String, String, int, String, boolean)}
+	 * instead
+	 */
+	@Deprecated
```

Comment:
I think we can use `@Deprecated(forRemoval = true, since = "6.1")` to add more detail to it.

Reply:
ok! I'll add more detail in `@Deprecated` annotation.